tools/dev/dockerpkg.py

```diff
@@ -43,30 +79,30 @@
 
 # Add core & aws packages
 ADD pyproject.toml poetry.lock README.md /src/
-ADD c7n /src/c7n/
 RUN . /usr/local/bin/activate && pip install -U pip
-RUN . /usr/local/bin/activate && poetry install --no-dev
+
+# Ignore root first pass so if source changes we don't have to invalidate
+# dependency install
+RUN . /usr/local/bin/activate && poetry install --without dev --no-root
 RUN . /usr/local/bin/activate && pip install -q wheel && \
       pip install -U pip
 RUN . /usr/local/bin/activate && pip install -q aws-xray-sdk psutil jsonpatch
 
+ARG providers="{providers}"
 # Add provider packages
+{PHASE_1_PKG_INSTALL}
+
+# Now install the root package
+ADD c7n /src/c7n/
+RUN . /usr/local/bin/activate && poetry install --only-root
+
+{PHASE_2_PKG_INSTALL}
+
 ADD tools/c7n_gcp /src/tools/c7n_gcp
-RUN rm -R tools/c7n_gcp/tests
```

Comment:
since the add is now specific to the phase2 these are duplicate afaics.

also nit let's rename the phase 1 / phase2 to suffix with install_pkg_dep / install_pkg_root for clarity

Reply:
Done!